network/limitcaller/rateLimitingTransport.go

```diff
@@ -54,14 +64,30 @@ func makeRateLimitingTransport(phonebook Phonebook, queueingTimeout time.Duratio
 	}
 }
 
+// MakeRateLimitingTransportWithTransport creates a rate limiting http transport that would limit the requests rate
+// according to the entries in the phonebook.
+func MakeRateLimitingTransportWithTransport(phonebook ConnectionTimeStore, queueingTimeout time.Duration, rt http.RoundTripper, target interface{}, maxIdleConnsPerHost int) RateLimitingTransport {
+	return RateLimitingTransport{
+		phonebook:       phonebook,
+		innerTransport:  rt,
+		queueingTimeout: queueingTimeout,
+		targetAddr:      target,
+	}
+}
+
 // RoundTrip connects to the address on the named network using the provided context.
 // It waits if needed not to exceed connectionsRateLimitingCount.
-func (r *rateLimitingTransport) RoundTrip(req *http.Request) (res *http.Response, err error) {
+func (r *RateLimitingTransport) RoundTrip(req *http.Request) (res *http.Response, err error) {
 	var waitTime time.Duration
 	var provisionalTime time.Time
 	queueingDeadline := time.Now().Add(r.queueingTimeout)
+	var host interface{} = req.Host
+	if len(req.Host) == 0 && req.URL != nil && len(req.URL.Host) == 0 {
```

Comment:
I think this if block is about identifying the P2P case - may want to explicitly comment that here (its implied in the line below however).

Reply:
moved the comment line up